src/quo2/components/list_items/messages_home_item.cljs

```diff
@@ -0,0 +1,136 @@
+                  {:components [rn/text]
+                   :length     0}
+                  children)
+
+                 "mention"
+                 {:components [rn/text @(re-frame/subscribe [:contacts/contact-name-by-identity literal])]
+                  :length     4} ;; we can't predict name length so take the smallest possible
+
+                 "status-tag"
+                 (truncate-literal (str "#" literal))
+
+                 "link"
+                 (truncate-literal destination)
+
+                 (truncate-literal literal))]
+    {:components (conj (:components acc) (:components result))
+     :length     (+ (:length acc) (:length result))}))
+
+(defn render-subheader
+  "Render the preview of a last message to a maximum of max-subheader-length characters"
+  [parsed-text]
+  (let [result
+        (reduce
+         (fn [{:keys [_ length] :as acc-text} new-text-chunk]
+           (if (>= length max-subheader-length)
+             (reduced acc-text)
+             (add-parsed-to-subheader acc-text new-text-chunk)))
+         {:components [rn/text {:style               (merge typography/paragraph-2 typography/font-regular
+                                                            {:color (colors/theme-colors colors/neutral-50 colors/neutral-40)
+                                                             :width "90%"})
+                                :number-of-lines     1
+                                :ellipsize-mode      :tail
+                                :accessibility-label :chat-message-text}]
+          :length     0}
+         parsed-text)]
+    (:components result)))
+
+(defn messages-home-item [item]
+  (let [{:keys [chat-id color group-chat last-message timestamp name]} item
+        display-name (if-not group-chat (first (<sub [:contacts/contact-two-names-by-identity chat-id])) name)
+        contact      (if-not group-chat (<sub [:contacts/contact-by-address chat-id]) nil)
+        photo-path   (if-not (empty? (:images contact)) (<sub [:chats/photo-path chat-id]) nil)]
+    ; TODO: community chats are not yet implemented. Some backend implementations are needed first. Also designs are not yet available.
+    [rn/touchable-opacity (merge {:style         {:margin-top         8
+                                                  :margin-horizontal  8
+                                                  :padding-vertical   8
+                                                  :padding-horizontal 12
+                                                  :border-radius      12
+                                                  :flex-direction     :row
+                                                  :align-items        :center}
+                                  :on-press      (fn []
+                                                   (re-frame/dispatch [:dismiss-keyboard])
+                                                   (if platform/android?
```

Comment:
Just food for thought (no specific suggestion for change in this PR), but whenever I see a check for `platform/android?` (or any platform check for that matter), I wonder if we should push such checks to the implementation side and not the caller side.

Reply:
What do you mean exactly? (I did not write this particluar on-press function, so I am not sure why is it needed)

Reply:
I'll try to expand @OmarBasem, but it's definitely not something your PR should attempt to improve necessarily, as this pattern is seen in other parts of the codebase.

Every time we inline platform checks, like `platform/android?` we run the risk of other places that forget to do this check to behave in unexpected ways. For example, for this particular line, if some other part of the code decides to dispatch `:chat.ui/navigate-to-chat-nav2` and forgets to put the platform guard clause, it'll be incorrectly dispatched for iOS. In this way, one could argue that it's the implementation responsibility to decide what to do, and not the caller that should know a hidden implementation detail and decide if it works in iOS or not. Of course there are exceptions.

In other words, when I read `(if platform/android? (dispatch [:A]) (dispatch [:B]))` I'm left wandering, why `:A` only works in Android? It reads to me like a hidden implementation detail leaking to the outside.

IMO what we should be building instead, is an explicit interface for things that behave differently between platforms. There are many other interesting examples in our codebase that would make my point easier to understand. If my examples are not clear enough, let's remember to chat about it next week!

I think we'll get back to this discussion little by little :)

Reply:
Thanks for the extensive explanation @ilmotta.